engine/engine/src/engine.cpp

```diff
@@ -394,19 +394,32 @@ namespace dmEngine
         if (engine->m_GuiContext)
             dmGui::DeleteContext(engine->m_GuiContext, engine->m_GuiScriptContext);
 
+        ScopedExtensionParams extension_params(engine);
         if (engine->m_SharedScriptContext) {
+            extension_params.SetLuaContext(engine->m_SharedScriptContext);
+            dmExtension::Finalize(extension_params);
```

Comment:
In 1.9.8, we've removed the old "scriptextension" concept, and made sure they're separate so it's easier to follow the life cycle of an extension. And technically, an extension doesn't _have_ to add a Lua context, which made the hard connection between scripts and extensions a bit weird.

And, during this refactor, I forgot to verify that the "Finalize" function is called.